Make TestPythonRegistration clean up after itself

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #102293
* __->__ #102292

We did this for TestCustomOp, now we are applying the same thing to
TestPythonRegistration.

This PR:
- changes TestPythonRegistration to register new ops under a single
namespace (self.test_ns)
- clean up the namespace by deleting it from torch.ops after each test
is done running.

This avoids a problem where if an op is re-defined, torch.ops.myns.op
crashes because we do some caching. The workaround in many of these
tests have been to just create an op with a different name, but this PR
makes it so that we don't need to do this.

Test Plan:
- existing tests

zou3519 committed:
Make TestPythonRegistration clean up after itself

We did this for TestCustomOp, now we are applying the same thing to
TestPythonRegistration.

This PR:
- changes TestPythonRegistration to register new ops under a single
namespace (self.test_ns)
- clean up the namespace by deleting it from torch.ops after each test
is done running.

This avoids a problem where if an op is re-defined, torch.ops.myns.op
crashes because we do some caching. The workaround in many of these
tests have been to just create an op with a different name, but this PR
makes it so that we don't need to do this.

Test Plan:
- existing tests

[ghstack-poisoned]

ezyang commented on test/test_python_dispatch.py:
```diff
@@ -399,14 +406,14 @@ def test_returning_symint(self) -> None:
         s0, s1 = ft.shape
 
-        tlib = Library("tlib", "DEF")
+        tlib = Library(f"{self.test_ns}", "DEF")
```
This f-string is useless, fwiw

zou3519 replied:
What do you mean?

zou3519 replied:
Nvm, I got it

zou3519 committed:
Update on "Make TestPythonRegistration clean up after itself"


We did this for TestCustomOp, now we are applying the same thing to
TestPythonRegistration.

This PR:
- changes TestPythonRegistration to register new ops under a single
namespace (self.test_ns)
- clean up the namespace by deleting it from torch.ops after each test
is done running.

This avoids a problem where if an op is re-defined, torch.ops.myns.op
crashes because we do some caching. The workaround in many of these
tests have been to just create an op with a different name, but this PR
makes it so that we don't need to do this.

Test Plan:
- existing tests

[ghstack-poisoned]
